src/compiler/checker.ts

```diff
@@ -9015,49 +9020,58 @@ namespace ts {
             return type;
         }
 
-        function resolveTypeOfAccessors(symbol: Symbol) {
+        function resolveTypeOfAccessors(symbol: Symbol, writing = false) {
             const getter = getDeclarationOfKind<AccessorDeclaration>(symbol, SyntaxKind.GetAccessor);
             const setter = getDeclarationOfKind<AccessorDeclaration>(symbol, SyntaxKind.SetAccessor);
 
+            // For write operations, prioritize type annotations on the setter

```

Comment:
also minor: a jsdoc description for the function is more readable than a stream of comments that happen right before the line of code that does what the comment says.